src/main/java/com/scalar/db/storage/cosmos/SelectStatementHandler.java

```diff
@@ -61,32 +67,48 @@ public SelectStatementHandler(CosmosClient client, TableMetadataManager metadata
     return Arrays.asList(record);
   }
 
+  private List<Record> executeReadWithIndex(Operation operation) throws CosmosException {
+    String query = makeQueryWithIndex(operation);
+    CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+    CosmosPagedIterable<Record> iterable =
+        getContainer(operation).queryItems(query, options, Record.class);
+
+    return Lists.newArrayList(iterable);
+  }
+
   private List<Record> executeQuery(Operation operation) throws CosmosException {
     CosmosOperation cosmosOperation = new CosmosOperation(operation, metadataManager);
     cosmosOperation.checkArgument(Scan.class);
     Scan scan = (Scan) operation;
 
-    String concatenatedPartitionKey = cosmosOperation.getConcatenatedPartitionKey();
-    SelectConditionStep<org.jooq.Record> select =
-        DSL.using(SQLDialect.DEFAULT)
-            .selectFrom("Record r")
-            .where(DSL.field("r.concatenatedPartitionKey").eq(concatenatedPartitionKey));
-
-    setStart(select, scan);
-    setEnd(select, scan);
-
-    setOrderings(select, scan.getOrderings());
+    String query;
+    CosmosQueryRequestOptions options;
+    if (Utility.isSecondaryIndexSpecified(scan, metadataManager.getTableMetadata(operation))) {
+      query = makeQueryWithIndex(scan);
```

Comment:
It looks like this doesn't consider Scan's orderings when a secondary index is specified?

Reply:
Thank you for the question.
That's because Cassandra doesn't support ordering by a secondary index.
I'll add a guard in `checkScanOperation`.

Reply:
> That's because Cassandra doesn't support ordering by a secondary index.

Oh, I didn't know that. Thank you!